lib/memoist.rb

```diff
@@ -203,21 +203,21 @@ def #{method_name}(reload = false)
           # end
 
           module_eval <<-EOS, __FILE__, __LINE__ + 1
-            def #{method_name}(*args)
+            def #{method_name}(*args, **kwargs)
               reload = Memoist.extract_reload!(method(#{unmemoized_method.inspect}), args)
 
-              skip_cache = reload || !(instance_variable_defined?(#{memoized_ivar.inspect}) && #{memoized_ivar} && #{memoized_ivar}.has_key?(args))
+              skip_cache = reload || !(instance_variable_defined?(#{memoized_ivar.inspect}) && #{memoized_ivar} && #{memoized_ivar}.has_key?(args+kwargs.values))
```

Comment:
ignoring of `kwargs.keys` is not quite accurate, this way we make no difference between `account(local: true)` and `account(remote: true)`

Reply:
True. Appreciate any help.

Reply:
what about using `kwargs.to_a`?

Reply:
we simply use `cache_key = args + [kwargs]` at the moment )